text_generation/causal_lm/cpp/continuous_batching/library/src/sampler.hpp

```diff
@@ -452,7 +463,9 @@ SamplerOutput Sampler::sample(std::vector<SequenceGroup::Ptr> & sequence_groups,
 
                 // check max length stop criteria
                 std::vector<Sequence::Ptr> running_sequences = sequence_group->get_running_sequences();
-                if (!sequence_group->has_finished() && running_sequences[0]->get_generated_len() == sampling_params.max_new_tokens) {
+                if (!sequence_group->has_finished() &&
+                    running_sequences[0]->get_generated_len() == sampling_params.max_new_tokens &&
+                    running_sequences[0]->get_generated_len() >= sampling_params.min_new_tokens) {
```

Comment:
I suppose `running_sequences[0]->get_generated_len() >= sampling_params.min_new_tokens` is always `true`, because `max_new_tokens >= min_new_tokens` 

I suppose if we apply `EOSPenaltyTransform` for beam search as well, it will be enough. But maybe we can do it separately when we will support all other penalties to beam search?